charms/kubernetes_snaps.py

```diff
@@ -192,7 +194,36 @@ def configure_apiserver(
     else:
         remove_if_exists(audit_webhook_conf_path)
 
-    configure_kubernetes_service("kube-apiserver", api_opts, extra_args_config)
+    config_files = set(
+        map(
+            api_opts.get,
+            (
+                "tls-cert-file",
+                "tls-private-key-file",
+                "kubelet-certificate-authority",
+                "kubelet-client-certificate",
+                "kubelet-client-key",
+                "authentication-token-webhook-config-file",
+                "service-account-signing-key-file",
+                "service-account-key-file",
+                "encryption-provider-config",
+                "etcd-cafile",
+                "etcd-keyfile",
+                "etcd-certfile",
+                "authorization-webhook-config-file",
+                "requestheader-client-ca-file",
+                "proxy-client-cert-file",
+                "proxy-client-key-file",
+                "client-ca-file",
+                "audit-policy-file",
+                "audit-webhook-config-file",
+            ),
+        )
+    )
```

Comment:
these keys point to config files that may have been overwritten during the reconcile period for the api server

Reply:
I know... that's what i'm SO afraid of.  Missing one would be bad

Reply:
@eaudetcobello do you have any ideas/opinions on how to not miss a file?  I could "scan all the arguments looking for file like strings" but that sounds not great

Reply:
the full list of args that can be files are listed here:
https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

Some with comments like:
`File with`
`Path to`
`File containing`
`file used`

And the hits just keep coming.  The docs and arguments names aren't consistent.  Maybe in the go code there's a list -- but with ever k8s release -- more and more arguments may or may not be config files.  Yes -- each config file will need to be ensured it gets added to the list and the regex is cool, but there's no guarantee upstream k8s will follow it.

Reply:
actually, there's another argument here expressly excluded b/c it's not a config file but a log file:
```audit-log-path```

we do not want to watch for changes in this file b/c it's a log.